aquacomputer_d5next.c

```diff
@@ -833,6 +853,67 @@ static int aqc_read_string(struct device *dev, enum hwmon_sensor_types type, u32
 	return 0;
 }
 
+static u8 leakshield_report_template[] = {
+	0x4, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
+};
+
+
+static int aqc_leakshield_send_report(struct aqc_data *priv, int channel, long val)
+{
+	struct usb_interface *intf;
+	struct usb_device *usb_dev;
+	unsigned int pipe;
+	int actual_length;
+	int ret;
+	u16 checksum;
+	u16 val16;
+
+	if (priv->kind != leakshield) {
+		return -EINVAL;
+	}
+
+	/* map -1 to N/A and forbid out-of-bounds values */
+	if (val == -1)
+	    /* note: the device will still remember the old value for 5 minutes */
+		val16 = 0x7fff;
+	else if (val < 0 || val >= 0x7fff)
+		return -EINVAL;
+	else
+		val16 = (u16)val;
+
+	/* leakshield_report_template is loaded into priv->buffer during initialization.
+	   if userland provides e.g. pump RPM and then flow rate, we don't want the flow
+	   rate update to clear the pump RPM, so we just keep the values saved in the buffer */
+
+	switch (channel) {
+	case 1:
+		put_unaligned_be16(val16, priv->buffer + 1);
+		break;
+	case 2:
+		put_unaligned_be16(val16, priv->buffer + 3);
```

Comment:
Same here with the `3`. I'd name it `LEAKSHIELD_USB_REPORT_FLOW_RATE_OFFSET` or similar.